include/aspect/initial_temperature/function.h

```diff
@@ -78,6 +80,13 @@ namespace aspect
          * A function object representing the temperature.
          */
         Functions::ParsedFunction<dim> function;
+
+        /**
+         * The coordinate representation to evaluate the function. Possible
+         * choices are depth, cartesian and spherical.
+         */
+        ::aspect::Utilities::Coordinates::CoordinateSystem coordinate_system;
```

Comment:
I don't understand this. If it compiles, it must work. Can you post what error message or fault you get?